core/src/test/java/org/infinispan/api/APINonTxTest.java

```diff
@@ -661,4 +677,99 @@ public boolean equals(Object obj) {
       }
    }
 
+   void assertLockStream(BiConsumer<Cache<Object, Object>, CacheEntry<Object, Object>> consumer) throws Throwable {
+      for (int i = 0; i < 10; i++) {
+         cache.put(i, "value" + i);
+      }
+
+      CyclicBarrier barrier = new CyclicBarrier(2);
+
+      int key = 4;
+
+      LockedStream<Object, Object> stream = cache.getAdvancedCache().lockedStream();
+      Future<?> forEachFuture = fork(() -> stream.forEach((c, e) -> {
+         if (e.getKey().equals(key)) {
+            try {
+               barrier.await(10, TimeUnit.SECONDS);
```

Comment:
> if a consumer is currently running

You don't block the consumer somewhere in the middle - most likely you'll run the operations in one order or in the other, but non-deterministically, which ain't good for the testsuite.